prow: cosi: add COSI Driver Sample and COSI monorepo config

shanduur opened the pull request:
Adding configuration for [COSI Driver Sample](https://github.com/kubernetes-sigs/cosi-driver-sample) and [COSI](https://github.com/kubernetes-sigs/container-object-storage-interface) monorepo build/test process. 

X-Ref: https://github.com/kubernetes-sigs/cosi-driver-sample/pull/13

shanduur commented:
@saad-ali @msau42 Could you please advise how to handle the retired repositories?

shanduur commented:
/assign @saad-ali 
/assign @msau42

BlaineEXE commented on config/jobs/kubernetes-sigs/container-object-storage-interface/container-object-storage-interface-api.yaml:
```diff
@@ -1,3 +1,4 @@
+# TODO(retired): should we drop this?
```
From meeting today, we can drop references to old repos

saad-ali commented:
Looks like some of the jobs are failing:

> could not write config: 3 errors occurred:
>	* could not find the referenced (Dashboard) sig-storage-container-object-storage-interface-api
>	* could not find the referenced (Dashboard) sig-storage-container-object-storage-interface-controller
>	* could not find the referenced (Dashboard) sig-storage-container-object-storage-interface-provisioner-sidecar
>	FAIL	k8s.io/test-infra/config/tests/testgrids	3.367s



CC @jsafrane @xing-yang

shanduur commented:
@saad-ali those are probably failing, because I removed the files (and added few new), cleaning those now. 
* _sig-storage-container-object-storage-interface-api_ 
* _sig-storage-container-object-storage-interface-controller_
* _sig-storage-container-object-storage-interface-provisioner-sidecar_

xing-yang commented:
I see that the issue is fixed now.

/lgtm
/approve

xing-yang commented:
@saad-ali Can you please help approve this?

saad-ali commented:
/lgtm
/approve
